src/app/components/CoinChart.tsx

```diff
@@ -87,13 +99,52 @@ export default function CoinChart({
       const { payload } = e.activePayload[0];
       const dataKey = chartType === "price" ? "value" : "volume";
       const value = payload[dataKey]?.toFixed(2) ?? "0";
+      const date = new Date(payload.date);
       setCurrentValue(value);
-      setCurrentDate(payload.date);
+      setCurrentDate(
+        date.toLocaleDateString(undefined, {
+          month: "long",
+          day: "numeric",
+          year: "numeric",
+        })
+      );
+    }
+  };
+
+  const formatDateGraphs = (tick: string): string => {
+    const date = new Date(tick);
+    let formattedDate: string;
+
+    if (days === "1") {
+      formattedDate = date.toLocaleTimeString(undefined, {
+        hour: "2-digit",
+        minute: "2-digit",
+        hour12: false,
+      });
+    } else if (["7", "14", "30"].includes(days)) {
+      formattedDate = date.toLocaleDateString(undefined, {
+        day: "numeric",
+        month: "short",
+      });
+    } else if (["90", "180", "365", "max"].includes(days)) {
+      formattedDate = date.toLocaleDateString(undefined, {
+        day: "numeric",
+        month: "short",
+        year: "numeric",
+      });
+    } else {
+      formattedDate = date.toLocaleDateString();
```

Comment:
This could be moved to a utility file from where you can export formatDateGraphs function because it does not depend on a state, it is a pure JS function.

Reply:
Good one. It makes sense to move it to utility folder as separate function. It makes code so much cleaner. Doing it right now.